src/workerd/jsg/value.h

```diff
@@ -468,20 +469,25 @@ class StringWrapper {
       kj::Maybe<v8::Local<v8::Object>> parentObject) {
     // TODO(cleanup): Move to a HeaderStringWrapper in the api directory.
     v8::Local<v8::String> str = check(handle->ToString(context));
-    auto result =
-        ByteString(KJ_ASSERT_NONNULL(tryUnwrap(context, str, (kj::String*)nullptr, parentObject)));
-    if (str->ContainsOnlyOneByte()) {
-      auto buf = kj::heapArray<kj::byte>(str->Length() + 1);
-      str->WriteOneByte(context->GetIsolate(), buf.begin());
-      for (auto b: buf) {
-        if (b >= 128) {
-          result.warning = ByteString::Warning::CONTAINS_EXTENDED_ASCII;
-          break;
-        }
+    auto result = ByteString(KJ_ASSERT_NONNULL(
+        tryUnwrap(context, str, static_cast<kj::String*>(nullptr), parentObject)));
```

Comment:
nit: I wonder if you can simply `tryUnwrap<kj::String*>` rather than cast.

Reply:
Wdym? I couldn't follow

Reply:
Ok, I understand it right now. The following code:

```c++
auto result = ByteString(KJ_ASSERT_NONNULL(
        tryUnwrap<kj::String*>(context, str, nullptr, parentObject)));
```

Throws build errors:

```
bazel-out/k8-fastbuild/bin/src/workerd/jsg/_virtual_includes/jsg/workerd/jsg/value.h:473:9: error: member 'tryUnwrap' used before its declaration
  473 |         tryUnwrap<kj::String*>(context, str, nullptr, parentObject)));
      |         ^
      |         this->
bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/kj/_virtual_includes/kj/kj/debug.h:361:27: note: expanded from macro 'KJ_ASSERT_NONNULL'
  361 | #define KJ_ASSERT_NONNULL KJ_REQUIRE_NONNULL
      |                           ^
bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/kj/_virtual_includes/kj/kj/debug.h:301:42: note: expanded from macro 'KJ_REQUIRE_NONNULL'
  301 |     auto _kj_result = ::kj::_::readMaybe(value); \
      |                                          ^
bazel-out/k8-fastbuild/bin/src/workerd/jsg/_virtual_includes/jsg/workerd/jsg/value.h:454:25: note: member is declared here
  454 |   kj::Maybe<kj::String> tryUnwrap(v8::Local<v8::Context> context,
      |                         ^
bazel-out/k8-fastbuild/bin/src/workerd/jsg/_virtual_includes/jsg/workerd/jsg/value.h:466:25: note: member is declared here
  466 |   kj::Maybe<ByteString> tryUnwrap(v8::Local<v8::Context> context,
      |                         ^
bazel-out/k8-fastbuild/bin/src/workerd/jsg/_virtual_includes/jsg/workerd/jsg/value.h:473:9: error: no matching member function for call to 'tryUnwrap'
  473 |         tryUnwrap<kj::String*>(context, str, nullptr, parentObject)));
      |         ^~~~~~~~~~~~~~~~~~~~~~
bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/kj/_virtual_includes/kj/kj/debug.h:361:27: note: expanded from macro 'KJ_ASSERT_NONNULL'
  361 | #define KJ_ASSERT_NONNULL KJ_REQUIRE_NONNULL
      |                           ^
bazel-out/k8-fastbuild/bin/external/capnp-cpp/src/kj/_virtual_includes/kj/kj/debug.h:301:42: note: expanded from macro 'KJ_REQUIRE_NONNULL'
  301 |     auto _kj_result = ::kj::_::readMaybe(value); \
```

Reply:
Mostly because tryUnwrap is an overload, and it doesn't take a template argument.